backend/src/services/worker/worker.service.ts

```diff
@@ -0,0 +1,89 @@
+import { worker as workerRep } from '~/data/repositories/repositories';
+import {
+  EAMWorkerCreateRequestDto,
+  EAMWorkerResponseDto,
+  TokenPayload,
+} from '~/common/types/types';
+import { Worker as WorkerEntity } from './worker.entity';
+import { HttpCode } from '~/common/enums/http/http';
+import {
+  encrypt as encryptServ,
+  token as tokenServ,
+  master as masterServ,
+} from '~/services/services';
+import { InvalidCredentialsError } from '~/exceptions/invalid-credentials-error/invalid-credentials-error';
+
+type Constructor = {
+  workerRepository: typeof workerRep;
+  encryptService: typeof encryptServ;
+  tokenService: typeof tokenServ;
+  masterService: typeof masterServ;
+};
+
+class Worker {
+  #workerRepository: typeof workerRep;
+  #encryptService: typeof encryptServ;
+  #tokenService: typeof tokenServ;
+  #masterService: typeof masterServ;
+
+  constructor({
+    workerRepository,
+    encryptService,
+    tokenService,
+    masterService,
+  }: Constructor) {
+    this.#workerRepository = workerRepository;
+    this.#encryptService = encryptService;
+    this.#tokenService = tokenService;
+    this.#masterService = masterService;
+  }
+
+  public async getAll(): Promise<Array<EAMWorkerResponseDto>> {
+    return await this.#workerRepository.getAll();
+  }
+
+  public async create({
+    name,
+    password,
+    token,
+  }: EAMWorkerCreateRequestDto): Promise<EAMWorkerResponseDto> {
+    const workerByName = await this.#workerRepository.getByName(name);
+
+    if (workerByName) {
+      throw new InvalidCredentialsError({
+        status: HttpCode.BAD_REQUEST,
+        message: `Worker with name ${name} exist`,
+      });
+    }
+
+    const { userId } = this.#tokenService.decode<TokenPayload>(token);
+
+    const master = await this.#masterService.getMasterById(userId);
+
+    if (!master) {
+      throw new InvalidCredentialsError({
+        status: HttpCode.UNAUTHORIZED,
+        message: 'Master not Found',
+      });
+    }
+
+    const passwordSalt = await this.#encryptService.createSalt();
+    const passwordHash = await this.#encryptService.createHash(
+      password,
+      passwordSalt,
+    );
+
+    const worker = WorkerEntity.createNew({
+      name,
+      passwordHash: passwordHash,
+      passwordSalt: passwordSalt,
+      tenantId: master.tenantId,
+    });
+
+    await this.#workerRepository.create(worker);
+
+    return worker;
```

Comment:
If in DB an error happens, the client will get a response with worker which wasn't created, you should return that, what your repository method returns
`return this.#workerRepository.create(worker);`

Reply:
I'd like to avoid this approach. If worker cannot be created we want to throw a specific exception. For example WokerCannotBeCreatedException. I don't see any advantage in `return this.#workerRepository.create(worker);`